aicsimageio/readers/czi_reader.py

```diff
@@ -92,11 +104,19 @@ def __init__(
                 f"Received URI: {self._path}, which points to {type(self._fs)}."
             )
 
+        # Handle deprecated parameter
+        if chunk_by_dims is not None:
+            log.warning(
```

Comment:
I think it might be better to use a `DeprecationWarning`? [docs](https://docs.python.org/3/library/exceptions.html#DeprecationWarning)

Reply:
The reason I went with log here instead of DeprecationWarning was because of:

>
>
> Changed in version 3.2: DeprecationWarning is now ignored by default in addition to PendingDeprecationWarning.
>
> Changed in version 3.7: DeprecationWarning is once again shown by default when triggered directly by code in __main__.
>
>

https://docs.python.org/3/library/warnings.html#default-warning-filter

Thoughts?

Reply:
Woah, why would they hide it by default?? It seems like it does get displayed for all the versions we support, so maybe it's okay to use now? 🤔

My thought is primarily, there is specific machinery that exists around deprecation warnings (IDE hints, pytest failures, etc) that would warrant using that in this case to tie into said machinery. Explicit logging as a warning may work to make the message more prominent, but circumvents that machinery. I know a lot of folks use the "zope" library for deprecation, maybe we should look into that?

Reply:
Ahhhh IDE stuff makes sense. I will use DeprecationWarning for now. We can decide if it's bad before 4.1 imo.